src/snowflake/cli/plugins/nativeapp/codegen/snowpark/extension_function_utils.py

```diff
@@ -83,3 +85,102 @@ def ensure_all_string_literals(values: Sequence[str]) -> List[str]:
         A list with all values transformed to be valid string literals (as necessary).
     """
     return [ensure_string_literal(value) for value in values]
+
+
+class _FunctionDefAccumulator(ast.NodeVisitor):
```

Comment:
I could use a docstring to understand how this visitor fits in. To me it looks like:

> Compiles a list of function definitions that match a set of `NativeAppExtensionFunction`s.